### PR TITLE
Bugfix for polynomial learning rate schedule

### PR DESCRIPTION
**Pull request by timmoon10**
The [Keras implementation](https://keras.io/api/optimizers/learning_rate_schedules/polynomial_decay/) of the polynomial decay learning rate schedule looks like:
```
lr(step) = (start_lr - end_lr) * (1 - step / max_step) ^ (power) + end_lr
```
However, we implement it as as:
```
lr(step) = (lr(step-1) - end_lr) * (1 - step / max_step) ^ (power) + end_lr
```
We currently only use the polynomial decay learning rate for [CosmoFlow](https://arxiv.org/pdf/1808.04728.pdf), which uses the Keras equation, so I think this is a bug.

I've modified the LBANN implementation match Keras and added a unit test. [Bamboo is running](https://lc.llnl.gov/bamboo/browse/LBANN-TIM418-1).

### CODE REVIEW

**Review by bvanessen — APPROVED** [review]

**Comment by bvanessen** [comment]
@timmoon10 Can you please take a look at this.  I just rebased it to address recently merged PRs.  Visually this looks good and I am recompiling it.

**Comment by bvanessen** [comment]
@timmoon10 It rebuilt for me.

**Comment by timmoon10** [comment]
[Bamboo is running](https://lc.llnl.gov/bamboo/browse/LBANN-TIM418-2).

**Comment by bvanessen** [comment]
@timmoon10 It also passed the local test for me.
